app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -0,0 +1,555 @@
+    getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+    toolbar.setNavigationOnClickListener(v -> onBackPressed());
+  }
+
+  private void setupLocationServices() {
+    LocationRequest locationRequest = new LocationRequest();
+    locationRequest.setInterval(10);
+    locationRequest.setSmallestDisplacement(10);
+    locationRequest.setFastestInterval(10);
+    locationRequest.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
+    LocationSettingsRequest.Builder builder = new
+        LocationSettingsRequest.Builder();
+    builder.addLocationRequest(locationRequest);
+
+    task = LocationServices.getSettingsClient(this).checkLocationSettings(builder.build());
+
+    locationSettingsResponseBuilder();
+  }
+
+  private void locationSettingsResponseBuilder() {
+    task.addOnCompleteListener(task -> {
+      try {
+        LocationSettingsResponse response = task.getResult(ApiException.class);
+        // All location settings are satisfied. The client can initialize location
+        // requests here.
+
+        createHotspotIntent(ACTION_TURN_ON_AFTER_O);
+
+        //}
+      } catch (ApiException exception) {
+        switch (exception.getStatusCode()) {
+          case LocationSettingsStatusCodes.RESOLUTION_REQUIRED:
+            // Location settings are not satisfied. But could be fixed by showing the
+            // user a dialog.
+            try {
+              // Cast to a resolvable exception.
+              ResolvableApiException resolvable = (ResolvableApiException) exception;
+              // Show the dialog by calling startResolutionForResult(),
+              // and check the result in onActivityResult().
+              resolvable.startResolutionForResult(
+                  ZimHostActivity.this,
+                  LOCATION_SETTINGS_PERMISSION_RESULT);
+            } catch (IntentSender.SendIntentException e) {
+              // Ignore the error.
+            } catch (ClassCastException e) {
+              // Ignore, should be an impossible error.
+            }
+            break;
+          case LocationSettingsStatusCodes.SETTINGS_CHANGE_UNAVAILABLE:
+            // Location settings are not satisfied. However, we have no way to fix the
+            // settings so we won't show the dialog.
+            break;
+          default:
+            break;
+        }
+      }
+    });
+  }
+
+  //Advice user to turn on hotspot manually for API<26
+  private void startHotspotManuallyDialog() {
+    AlertDialog.Builder builder = new AlertDialog.Builder(this, dialogStyle());
+
+    builder.setPositiveButton(getString(R.string.go_to_settings_label),
+        (dialog, id) -> launchTetheringSettingsScreen());
+
+    builder.setNeutralButton(getString(R.string.hotspot_dialog_neutral_button), (dialog, id) -> {
+
+      progressDialog =
+          ProgressDialog.show(this, getString(R.string.progress_dialog_starting_server), "",
+              true);
+      progressDialog.show();
+
+      startFlowable();
```

Comment:
what Flowable, what does it do? rename apropriately

Reply:
Does `isIpAddressValid` seems like a better name? fdca8057172c67ee73dac7b1de8b683513cf9289

Reply:
not really what is happening, `pollForValidIpAddress` is a more descriptive about what your are doing

Reply:
f05e254043738d86108e4358bba79fc06eeef896